posthog/hogql_queries/experiments/test/test_experiment_trend_query_runner.py

```diff
@@ -338,3 +330,248 @@ def test_query_runner_with_avg_math(self):
+        experiment = self.create_experiment(feature_flag=feature_flag)
+
+        ff_property = f"$feature/{feature_flag.key}"
+        journeys_for(
+            {
+                "user_control": [
+                    {"event": "$pageview", "timestamp": "2020-01-02", "properties": {ff_property: "control"}},
+                ],
+            },
+            self.team,
+        )
+
+        flush_persons_and_events()
+
+        count_query = TrendsQuery(series=[EventsNode(event="$pageview")])
+        experiment_query = ExperimentTrendQuery(
+            experiment_id=experiment.id,
+            kind="ExperimentTrendQuery",
+            count_query=count_query,
+        )
+
+        query_runner = ExperimentTrendQueryRunner(query=experiment_query, team=self.team)
+        result = query_runner.count_query_runner.calculate()
+
+        with self.assertRaises(ValidationError) as context:
+            query_runner._validate_event_variants(result)
+
+        expected_errors = json.dumps(
+            {
+                ExperimentNoResultsErrorKeys.NO_EVENTS: False,
+                ExperimentNoResultsErrorKeys.NO_FLAG_INFO: False,
+                ExperimentNoResultsErrorKeys.NO_CONTROL_VARIANT: False,
+                ExperimentNoResultsErrorKeys.NO_TEST_VARIANT: True,
+            }
+        )
+        self.assertEqual(cast(list, context.exception.detail)[0], expected_errors)
+
+    @freeze_time("2020-01-01T12:00:00Z")
+    def test_validate_event_variants_no_flag_info(self):
+        feature_flag = self.create_feature_flag()
+        experiment = self.create_experiment(feature_flag=feature_flag)
+
+        journeys_for(
+            {
+                "user_no_flag_1": [
+                    {"event": "$pageview", "timestamp": "2020-01-02"},
+                ],
+                "user_no_flag_2": [
+                    {"event": "$pageview", "timestamp": "2020-01-03"},
+                ],
+            },
+            self.team,
+        )
+
+        flush_persons_and_events()
+
+        count_query = TrendsQuery(series=[EventsNode(event="$pageview")])
+        experiment_query = ExperimentTrendQuery(
+            experiment_id=experiment.id,
+            kind="ExperimentTrendQuery",
+            count_query=count_query,
+        )
+
+        query_runner = ExperimentTrendQueryRunner(query=experiment_query, team=self.team)
+        result = query_runner.count_query_runner.calculate()
+
+        with self.assertRaises(ValidationError) as context:
+            query_runner._validate_event_variants(result)
+
+        expected_errors = json.dumps(
+            {
+                ExperimentNoResultsErrorKeys.NO_EVENTS: True,
+                ExperimentNoResultsErrorKeys.NO_FLAG_INFO: True,
+                ExperimentNoResultsErrorKeys.NO_CONTROL_VARIANT: True,
+                ExperimentNoResultsErrorKeys.NO_TEST_VARIANT: True,
+            }
+        )
+        self.assertEqual(cast(list, context.exception.detail)[0], expected_errors)
```

Comment:
are all the tests copied here from the old trends query?

Reply:
The shared stats methods are all unit-tested here: https://github.com/PostHog/posthog/blob/9a7deda3fcdd62da7cdafb7bebbe441fa3a64d82/ee/clickhouse/queries/experiments/test_trend_experiment_result.py#L4

Other than that, the validation errors that are tested in https://github.com/PostHog/posthog/blob/9a7deda3fcdd62da7cdafb7bebbe441fa3a64d82/ee/clickhouse/queries/test/test_experiments.py#L4 were ported here.